src/services/consumer-view.ts

```diff
@@ -0,0 +1,95 @@
+import { getFileService } from '../utils/get-file-service';
+import { Revision } from '../entities/dataset/revision';
+import { validateParams } from '../validators/preview-validator';
+import { CSVHeader, ViewDTO, ViewErrDTO } from '../dtos/view-dto';
+import { FactTableColumnType } from '../enums/fact-table-column-type';
+import { DatasetDTO } from '../dtos/dataset-dto';
+import { Dataset } from '../entities/dataset/dataset';
+
+export const getTableRowsNoFilter = async (
+  datasetId: string,
+  revision: Revision,
+  lang: string,
+  startRow: number,
+  endRow: number,
+  sortBy?: string
+) => {
+  const { parquetMetadata, parquetQuery } = await import('hyparquet');
+  const fileService = getFileService();
+  const stream = await fileService.loadStream(`${revision.id}_${lang}.parquet`, datasetId);
+  const chunks = [];
+  for await (const chunk of stream) {
+    chunks.push(chunk);
+  }
+  const buffer = Buffer.concat(chunks);
+  const arrayBuffer = buffer.buffer.slice(buffer.byteOffset, buffer.byteOffset + buffer.byteLength);
+
+  const asyncBuffer = {
+    byteLength: arrayBuffer.byteLength,
+    slice: async (start: number | undefined, end: number | undefined) => arrayBuffer.slice(start, end)
+  };
+
+  const metadata = parquetMetadata(arrayBuffer);
+  const rows = await parquetQuery({ file: asyncBuffer, orderBy: sortBy, rowStart: startRow, rowEnd: endRow });
+
+  return { metadata, rows };
+};
+
+export const getTableMetadata = async (datasetId: string, revision: Revision, lang: string) => {
+  const { parquetMetadata } = await import('hyparquet');
+  const fileService = getFileService();
+  const stream = await fileService.loadStream(`${revision.id}_${lang}.parquet`, datasetId);
```

Comment:
Same as above, can we use `loadBuffer()` ?

Reply:
Yeah... Switching to buffer worked and saved 6 lines of code